vendor/github.com/vmware/gvddk/gvddk-high/gvddk.go

```diff
@@ -330,4 +330,9 @@ func (this DiskConnectHandle) Close() error {
 
 func (this DiskConnectHandle) Capacity() int64 {
 	return int64(this.info.Capacity) * gDiskLib.VIXDISKLIB_SECTOR_SIZE
-}
\ No newline at end of file
+}
+
+// QueryAllocatedBlocks invokes the VDDK function of the same name.
+func (this DiskReaderWriter) QueryAllocatedBlocks(startSector gDiskLib.VixDiskLibSectorType, numSectors gDiskLib.VixDiskLibSectorType, chunkSize gDiskLib.VixDiskLibSectorType) ([]gDiskLib.VixDiskLibBlock, gDiskLib.VddkError) {
```

Comment:
Sorry for the late review.  Can we move this to DiskConnectHandle?  It doesn't seem like a Reader/Writer method.

Reply:
Thanks, what you say make sense! 

Would you object, though, if it still gets exposed via `DiskReaderWriter` along the lines of the `DiskReaderWrite.ReadAt()` and `DiskReaderWrite.WriteAt()` methods? e.g.
```
func (this DiskReaderWriter) QueryAllocatedBlocks(startSector gDiskLib.VixDiskLibSectorType, numSectors gDiskLib.VixDiskLibSectorType, chunkSize gDiskLib.VixDiskLibSectorType) ([]gDiskLib.VixDiskLibBlock, gDiskLib.VddkError) {
	return this.diskHandle.QueryAllocatedBlocks(startSector, numSectors, chunkSize)
}
```

The reason I ask is that the `gvddk_high.Open` method calls many `gDiskLib` methods internally, including `PrepareForAccess`. Besides being super-convenient, I don't know if one *can* open another handle on the same volume to just make the `QueryAllocatedBlocks` call.

Reply:
Sure, we can add it to both, thanks